internal/handlers/boot_artifacts.go

```diff
@@ -37,6 +37,12 @@ func parseArtifact(path, arch string) (string, error) {
 		} else {
 			artifact = "vmlinuz"
 		}
+	case "ins-file":
+		if arch == "s390x" {
+			artifact = "generic.ins"
+		} else {
+			return "", fmt.Errorf("artifact not available for architectures other than s390x. Current arch: %s", arch)
```

Comment:
Maybe just "ins file is only available for s390x architecture"?

Reply:
Yes, the generic.ins file is only available on s390x images. This is the root content for x86:
dr-xr-xr-x. 2 root root 2048 Feb 13 02:18 coreos
dr-xr-xr-x. 3 root root 2048 Feb 13 02:18 EFI
dr-xr-xr-x. 3 root root 2048 Feb 13 02:18 images
dr-xr-xr-x. 2 root root 2048 Feb 13 02:18 isolinux
-r--r--r--. 1 root root   89 Feb 13 01:20 zipl.prm

Reply:
Yes, I know. I was just suggesting a simpler error message.